Core.PoseEditor/AMModules/BonesEditor.cs

```diff
@@ -1241,19 +1246,40 @@ private static bool WildCardSearch(string text, string search)
             return Regex.IsMatch(text, regex, RegexOptions.IgnoreCase | RegexOptions.Compiled);
         }
 
-        private void DisplayObjectTree(GameObject go, int indent)
+        private void UpdateSearch()
         {
-            if (_parent._childObjects.Contains(go))
-                return;
-            string displayedName;
-            bool aliased = true;
-            if (_boneAliases.TryGetValue(go.name, out displayedName) == false)
+            searchResults = new List<GameObject>();
+            void FilterBones(GameObject go)
             {
-                displayedName = go.name;
-                aliased = false;
+                if (_parent._childObjects.Contains(go))
+                    return;
+                string displayedName;
+                bool aliased = true;
+                if (_boneAliases.TryGetValue(go.name, out displayedName) == false)
+                {
+                    displayedName = go.name;
+                    aliased = false;
+                }
+
+                if (_search.Length == 0 || WildCardSearch(go.name, _search) || (aliased && WildCardSearch(displayedName, _search)))
+                    searchResults.Add(go);
+
+                for (int i = 0; i < go.transform.childCount; ++i)
+                    FilterBones(go.transform.GetChild(i).gameObject);
+            }
+
+            foreach (Transform child in _parent.transform)
+            {
+                FilterBones(child.gameObject);
             }
+        }
+
+        private void DisplayObjectTree(GameObject go, int indent)
+        {
+            if (_boneAliases.TryGetValue(go.name, out string displayedName) == false)
+                displayedName = go.name;
 
-            if (_search.Length == 0 || WildCardSearch(go.name, _search) || (aliased && WildCardSearch(displayedName, _search)))
+            if (searchResults.Contains(go))
```

Comment:
if `_search.Length == 0` you could not call UpdateSearch at all and skip this Contains check. It will be faster, not sure how much though.